### DISCUSSION TITLE
How to output the assigned area to each node for integrated BCs?

### DISCUSSION THREAD

**Discussion by garciapintado** [idaholab/moose#22878] [Q&A Getting Started] [answered]
As I understand, when saved (`save_in=True`), the values from integrated boundary condition objects (e.g. `NeumannBC`) give the flux assigned to each boundary node in the considered sideset, where each node is assigned an area.
Please, is there any direct way (which I am then missing) to output the nodal area assigned to each  of these BC nodes so one could get the flux per unit area?

Thanks!

**Comment by GiudGiud** [comment]
I think the area around the node is baked into _JxW[0] for a kernel acting on a nodal variable.
It should be similar for a flux BC

Guillaume

**Comment by garciapintado** [comment]
Thanks for answer, @GiudGiud . Sorry that I am not getting this.  As I've understood, I should create a Boundary restricted Nodal AuxKernel object (e.g., let’s call it NodalAreaBC), inheriting from Auxkernel, and with computeValue as:
```
Real
NodalAreaBC::computeValue()
{
  if (isNodal())
      return _JxW[0];
  else
      return _JxW[_qp];
}
```
Which obviously is wrong as it does not work. That is, for an elemental auxiliary variable variable it outputs something proportional to the element size, but not quite the element area, and for a nodal auxiliary variable, as follows, it directly gives a segmentation error. 
```
[AuxVariables]
  [node_area]
    family = LAGRANGE
    order = FIRST
    #family = MONOMIAL
    #order = CONSTANT
  []
[]

[AuxKernels]
  [node_area]
    type = NodalAreaBC
    variable = node_area
    boundary = top
  []
[]
 ```
Could you please advise further?

**Reply by GiudGiud** [reply]
The behavior with the elemental variable is expected.
Do you need this output in an auxiliary variable? Or can you use it on the fly in a calculation?

Without digging into it, I dont think the auxkernel will have the local nodal area computed at any point. I would expect boundary conditions (which will not allow you to output the quantity) to have it

**Reply by garciapintado** [reply]
The point is that if I have an integratedBC for energy flux, whose output is, integrated for each node [J.s-1] through a boundary. This is nice as I can directly use also a postprocessor to get the nodal sum, and so the integrated energy flux [J.s-1] across all the boundary as a global value.

So far so good, but on the other hand, the element size along this boundary [with a rather high spatial variability] implies that plotting of this flux, as it is a product of the flux per unit area [J.m-2.s-1] ---what I would like to see in plots--- times the assigned nodal size [m2], is not so nice.

The only way I can think of doing this is by adding an Auxkernel, which would be calculated just once within MOOSE, indicating the corresponding nodal areas for the element boundaries. I could then later [out of MOOSE] divide the integratedBC saved_in value by the nodal areas.

Maybe I'm missing an alternative?

**Reply by GiudGiud** [reply]
Can you divide the integrated flux by the total surface area instead of the local nodal area?
The total area of a sideset is easy to get with a surface postprocessor

**Reply by garciapintado** [reply]
Well, no. That would give the average flux intensity [J.m-2.s-1], but not the spatial distribution of the intensity (of course, the same units [J.m-2.s-1]).

**Reply by garciapintado** [reply]
I've been for a few days trying to see if I was missing some MOOSE option for this. A "promising" object name was the auxkernel `NodalVolumePD`, from the peridynamics module , but it seems to be only for PD mesh objects.

**Comment by maxnezdyur** [comment, marked as answer]
There is an user object called `NodalArea`. It may be what you are looking for.

**Reply by garciapintado** [reply]
Right! In the `contact` module. It does not appear  listed in the Moose documentation Syntax Index, but it does within the documentation source list. I had completely missed it. I've just tried and works as expected! Many thanks!
